proposals/0016-simd-application-fees.md

```diff
@@ -0,0 +1,278 @@
+---
+simd: '0016'
+title: Application Fees (Write-lock fees)
+authors:
+  - Godmode Galactus (Mango Markets)
+  - Maximilian Schneider (Mango Markets)
+category: Standard
+type: Core, Fees
+status: Draft
+created: 2022-12-23
+feature:
+---
+
+
+## Summary
+
+This SIMD will discuss additional fees called Application Fees or write lock fees.
+These fees are decided and set by the dapp developer to interact with the dapp. 
+Dapp developers then can decide to rebate these fees if a user interacts with the dapp 
+as intended and disincentivize the users which do not.
+These fees will be applied even if the transaction eventually fails and collected on 
+the same writable account.
+Account authority (i.e owner of the account) can do lamport transfers to recover these fees.
+So instead of fees going to the validator these fees go to the dapp developers.
+
+Discussion for the issue : <https://github.com/solana-labs/solana/issues/21883>
+
+## Motivation
+
+During network congestion, many transactions in the network cannot be processed
+because they all want to write-lock same accounts. When a write lock on an account is
+taken by a transaction batch no other batch can use this account in parallel, so only
+transactions belonging to a single batch are processed correctly and all others are retried
+again or forwarded to the leader of the next slot. With all the forwarding and retrying the
+validator is choked by the transactions write-locking the same accounts and effectively
+processing valid transactions sequentially.
+
+There are multiple accounts of OpenBook (formerly serum), mango markets, etc which
+are used in high-frequency trading. During the event of extreme congestion, we observe
+that specialized trading programs write lock these accounts but never CPI into the respective
+programs unless they can extract profit, effectively starving the actual users for access.
+Penalizing this behavior with runtime fees collected by validators can create a perverse
+incentive to artificially delay HFT transactions, cause them to fail, and charge penalty fees.
+
+## Alternatives Considered
+
+* Having a fixed write lock fee and a read lock fee.
```

Comment:
what about dynamic write/read locked fee controlled by some math calculation in consensus (similar to bank timestamp or other consensus values)

Reply:
> what about dynamic write/read locked fee controlled by some math calculation in consensus (similar to bank timestamp or other consensus values)

can be complementary but does not fulfill the same goal, should be discussed in it's own SIMD